### ISSUE TITLE
Rectangle patch constructor fails with units

### ISSUE THREAD

**Issue by TD22057** [matplotlib/matplotlib#5421]
Using unit'ized inputs in Rectangle fails in the constructor:

```
    File "test_units_patches.py", line 23, in test_units_rectangle
        p = mpatches.Rectangle( ( 5*U.km, 6*U.km ), 1*U.km, 2*U.km )
    File "/home/trd/git/TD22057/matplotlib/build/lib.linux-x86_64-2.7/matplotlib/patches.py", line 639, in __init__
        self._x = float(xy[0])
    TypeError: float() argument must be a string or a number
```

This is a new problem with 1.5.

**Comment by TD22057** [comment]
Pull request #5422 adds a test case and correct image for this this failure but doesn't fix Rectangle.  This was changed for #4751 but no test case was added for that failure so there is no easy way to verify both fixes.

**Comment by dstansby** [comment]
Sorry for the long delay here; there's a small python3 compatibility bug in `jpl_units` that needs to be fixed, but after that and once #8938 is fixed this should be good to go.
